test/bitmasks.jl

```diff
@@ -14,8 +14,10 @@ end
     @test BIT_A & BIT_B == FlagBits(0)
     @test BIT_A | BIT_B == BIT_C
     @test xor(BIT_A, BIT_B) == BIT_C
+    @test isless(BIT_A, BIT_B) && isless(BIT_B, BIT_C) && isless(BIT_A, BIT_C)
```

Comment:
Just a matter of style, but it would be more readable with `<` instead of the plain `isless`.

Reply:
Do we want a partial order here? That's the distinction between `<` and `isless` on a semantic level in julia.

> <(x, y)                                                                                                                                             
>                                                                                                                                                    
> Less-than comparison operator. Falls back to isless. Because of the behavior of floating-point NaN values, this operator implements a partial order.

> help> isless
> [...]
>  Non-numeric types with a total order should implement this function. Numeric types only need to implement it if they have special values such as NaN. Types with a partial order should implement <. See the documentation on Alternate orderings for how to define alternate ordering methods that can be used in sorting and related functions.

Reply:
Or do you mean just in the test?

Reply:
Oh, I didn't know about this distinction. I meant in the test, and with this semantic difference it should definitely be just in tests (as you would use it in typical code)

Reply:
While reworking them, I've noticed that we'd also need `==` defined on `BitMask` and `Integer` for `Flag <= Integer` - should we do this as well or is that too much?

Reply:
I am not sure that equality or `<=` would be a good idea to compare between flags. For example, `DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT | DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT != DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT`

but comparisons with `<` make IMO more sense with

`DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT | DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT <  DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT`

Equality between `BitMask` was already defined by default, but I don't think we should use it. Please let me know what you think. If you see a clear use of `==` for flags we may add it, otherwise I propose to merge this PR.

Reply:
I'm not too sure either, the tutorial at least [seems to encourage it](https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Validation_layers) (scroll down a little or search for `messageSeverity >=`):

> The values of this enumeration are set up in such a way that you can use a comparison operation to check if a message is equal or worse compared to some level of severity, for example

The way I'd usually use equality would be in the sense of 

```
severity & DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT == DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT 
```

It also mirrors how LogLevels in julia work - if you specify `Info` as the minimum loglevel, you'd expect logging messages with level `Warning` or `Error` to show up, while those with level `Debug` would be suppressed.

So all in all, I think keeping the default behaviour is fine, except maybe we'd want to explicitly disallow equality comparison between different BitMasks.

Reply:
Hmm, in the case where we compare with `<` we're fine, as in `severity < DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT`, but if we compare with `>`, we'd need to `|` all combinations if we really want to pick "above" a certain flag. For example `severity > DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT` would be true if severity was `DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT | DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT`, which would be annoying and would force to compare to warning | info | verbose. The only practical possibility I see would be to use `>=` as in `severity >= DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT`.

So maybe we should commit to defining `==` after all. I couldn't see a use case for `<=` but now I see one for  `>=` as in the vulkan-tutorial. If you agree, then you may proceed to define `==` between integers and bitmasks, and revive the two inequality tests. The definition of an order on bitmasks is quite tricky, we may come back to that in the future.

> It also mirrors how LogLevels in julia work - if you specify Info as the minimum loglevel, you'd expect logging messages with level `Warning` or `Error` to show up, while those with level `Debug` would be suppressed.

True, though this is special to logging. The relation `>= Info` there implies `>= Info | Warning | Error`.

> The way I'd usually use equality would be in the sense of

Yeah, we'd need a handy operator for that, and equality is not really it. Maybe something like `contains(severity, DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT)`? Or `in`? You can open a separate issue on that if you want, to avoid cluttering this PR.

Reply:
> maybe we'd want to explicitly disallow equality comparison between different BitMasks.

This is already working, since it's checking for `===`, the difference in type makes it return false.

Reply:
> For example severity > DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT would be true if severity was DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT | DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT

Yeah, I agree that it kind of breaks down as soon as you `|` different levels together - does combining log levels like that even make sense on a conceptual level in Vulkan? Is that something people/devices do?

> Yeah, we'd need a handy operator for that, and equality is not really it. Maybe something like contains(severity, DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT)? Or in? You can open a separate issue on that if you want, to avoid cluttering this PR.

Yeah, logging is hard :/ I'll open an issue.

> This is already working as intended I think, since it's checking for ===, the difference in type makes it return false.

Right, but that doesn't catch that comparing different types doesn't make sense for BitMasks (as mentioned in the [manual](https://juliagpu.github.io/Vulkan.jl/stable/features/#Bitmask-flags) for this wrapper):

> To circumvent that, every bitmask flag now has one associated type which prevents combinations with flags of other bitmask types.

So personally I'd disallow it for consistencies sake. Doesn't have to be in this PR though. Since we're in 0.x anyway, we can break and just bump a minor version.

Reply:
> Yeah, I agree that it kind of breaks down as soon as you | different levels together - does combining log levels like that even make sense on a conceptual level in Vulkan? Is that something people/devices do?

For logging, I personally don't think it makes sense. It does for other bitmasks though (see #13).

> So personally I'd disallow it for consistencies sake. Doesn't have to be in this PR though. Since we're in 0.x anyway, we can break and just bump a minor version.

That's fair. Let's merge this first and we can come back to that after having given it some more thoughts.

Reply:
I'll open a new issue, so we don't forget about the equality between different bitmasks.